example/Setup.hs

```diff
@@ -1,6 +1,31 @@
-module Main where
+{-# LANGUAGE CPP #-}
+{-# OPTIONS_GHC -Wall #-}
+module Main (main) where
 
-import Distribution.Extra.Doctest (defaultMainWithDoctests)
+#ifndef MIN_VERSION_cabal_doctest
+#define MIN_VERSION_cabal_doctest(x,y,z) 0
+#endif
 
+#if MIN_VERSION_cabal_doctest(1,0,0)
+
+import Distribution.Extra.Doctest ( defaultMainWithDoctests )
 main :: IO ()
 main = defaultMainWithDoctests "doctests"
+
+#else
+
+#ifdef MIN_VERSION_Cabal
+-- If the macro is defined, we have new cabal-install,
```

Comment:
I suppose we could. The only quibbles I have with that idea are:

1. `#warning` isn't a standardized thing
2. We'd add even more CPP

In practice, neither of those are issues, since:

1. Pretty much every CPP preprocessor supports `#warning`, which includes `gcc`, `clang`, and `cpphs`. The only one I can think of that _doesn't_ support `#warning` is MSVC, but then again, nobody uses that with GHC :)
2. We're already using a fair bit of CPP in the first place, so... what's a little more going to hurt?

I'd be fine with implementing it either way.